lib/robots/dor_repo/accession/publish.rb

```diff
@@ -14,7 +14,12 @@ def perform(druid)
           object_client = Dor::Services::Client.object(druid)
           obj = object_client.find
 
-          object_client.publish unless obj.is_a?(Cocina::Models::AdminPolicy)
+          return if obj.is_a?(Cocina::Models::AdminPolicy)
+
+          background_result_url = Dor::Services::Client.object(druid).publish
+          result = Dor::Services::Client::AsyncResult.new(url: background_result_url)
```

Comment:
And I believe you can use the new interface here: `result = object_client.async_result(url: background_result_url)` (assuming that made it into a release)

Reply:
that hasn't been released

Reply:
now it has! 3.6.0